internal/controller/pipelinerollout_controller.go

```diff
@@ -145,6 +183,80 @@ func (r *PipelineRolloutReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	return ctrl.Result{}, nil
 }
 
+func (r *PipelineRolloutReconciler) Shutdown() {
+	numaLogger := logger.GetBaseLogger().WithName(loggerName)
```

Comment:
Why not getting the logger from context?

Reply:
Yes, that's probably good if I pass in `context`. We can always get it from `context` if we're not the highest function on the stack that we can write to.